core/src/main/java/io/stargate/core/activator/BaseActivator.java

```diff
@@ -280,15 +280,15 @@ protected void stopService() {
 
     private final Hashtable<String, String> properties;
 
-    public ServiceAndProperties(
-        Object service, Class<?> targetServiceClass, Hashtable<String, String> properties) {
+    public <T, E extends T> ServiceAndProperties(
+        E service, Class<T> targetServiceClass, Hashtable<String, String> properties) {
       this.service = service;
       this.targetServiceClass = targetServiceClass;
       this.properties = properties;
     }
 
     @SuppressWarnings("JdkObsolete")
-    public ServiceAndProperties(Object service, Class<?> targetServiceClass) {
+    public <T, E extends T> ServiceAndProperties(E service, Class<T> targetServiceClass) {
```

Comment:
Do we have to re-declared generics on the constructor (`<T, E extends T>`)?

Reply:
See above

Reply:
You're right. My mistake. I though the first type param was on the class declaration :facepalm: